### PR TITLE
[DDW-613] Fix "Send form" fee calculation edge-cases

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR fixed fee calculation edge cases in wallet send form.
[Jira Ticket](https://jira.iohk.io/browse/DDW-613)

## Todos
- [x] Fix fee calculation edge cases in wallet send form

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1617362437315300)

## Test Cases

```Gherkin
Scenario 1: Select additional ‘ADD TOKEN’ button before fees calculation finishes
Given I am on Send page
And I have inputted a valid receiver address
And I input ‘5’ ADA to send
And I have waited estimated fees to be calculated
And I have selected ‘ADD A TOKEN' button
When I input ‘1’ token to send
And before fee calculation finishes I select ‘ADD A TOKEN’ button
Then there shouldn’t be any error
And the behaviour is as expected


Scenario 2: Select ‘ADD TOKEN’ button before fees calculation finishes
Given I am on Send page
And I have inputted a valid receiver address
When I input ‘1’ ADA to send
And before fee calculation finishes I select ‘ADD A TOKEN’ button
Then there shouldn’t be any error
And the behaviour is as expected
```

### App
- [x] Go to `Wallet` => `Send` form and make sure fee calculation is reset after clicking on `Add token` button fast while ADA amount fee is being calculated

### English
App_en

### Japanese
App_jp

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by ManusMcCole — APPROVED** [review]

**Comment by ManusMcCole** [comment]
Looks good to me 👍

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Tested on 17503. Great work @yakovkaravelov

File: source/renderer/app/components/wallet/WalletSendForm.js [hunk 1]
```diff
@@ -454,7 +454,8 @@ export default class WalletSendForm extends Component<Props, State> {
         this.isLatestTransactionFeeRequest(
           this.state.feeCalculationRequestQue,
           prevFeeCalculationRequestQue
-        )
+        ) &&
+        !this.selectedAssetsAmounts.includes('0')
```
Comment (tomislavhoracek) [inline source/renderer/app/components/wallet/WalletSendForm.js:458]: @yakovkaravelov I'm not sure that this is correct. What will be if the asset amount is `10` ? Maybe you should check if the asset added and not empty.

Comment (yakovkaravelov) [reply source/renderer/app/components/wallet/WalletSendForm.js:458]: @tomislavhoracek Yeah, you are right. This conditional check was just copied from different place of this same file...
Need to fix this one and that one as well.

Comment (yakovkaravelov) [reply source/renderer/app/components/wallet/WalletSendForm.js]: @tomislavhoracek It is not a string, it is an array of string.
So `10` can't satisfy above condition.

Comment (tomislavhoracek) [reply source/renderer/app/components/wallet/WalletSendForm.js]: @yakovkaravelov oke, then we are good!

File: source/renderer/app/utils/formatters.js [hunk 2]
```diff
@@ -95,7 +95,11 @@ export const shortNumber = (value: number | BigNumber): string => {
   return formattedAmount;
 };
 
-export const formattedAmountToNaturalUnits = (amount: string): string => {
+export const formattedAmountToNaturalUnits = (amount: ?string): string => {
+  if (amount === undefined || amount === null) {
```
Comment (tomislavhoracek) [inline source/renderer/app/utils/formatters.js:99]: @yakovkaravelov can we replace this with `if (!amount)` here?

**Review by tomislavhoracek — APPROVED** [review]
💯

**Comment by tomislavhoracek** [comment]
@ManusMcCole @miorsufianiohk Can you guys check this one more time because @yakovkaravelov made some changes. Just to be sure that everything is still ok.

**Review by miorsufianiohk — APPROVED** [review]
LGTM 👍 . Great work @yakovkaravelov . Tested on 17538
